src/main/java/org/openrewrite/hibernate/MigrateUserType.java

```diff
@@ -0,0 +1,253 @@
+ * limitations under the License.
+ */
+package org.openrewrite.hibernate;
+
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.internal.ListUtils;
+import org.openrewrite.java.JavaParser;
+import org.openrewrite.java.JavaTemplate;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.*;
+import org.openrewrite.marker.Markers;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+import static org.openrewrite.Tree.randomId;
+
+public class MigrateUserType extends Recipe {
+
+    private static final String USER_TYPE = "org.hibernate.usertype.UserType";
+    private static final MethodMatcher SQL_TYPES = new MethodMatcher("* sqlTypes()");
+    private static final MethodMatcher RETURNED_CLASS = new MethodMatcher("* returnedClass()");
+    private static final MethodMatcher EQUALS = new MethodMatcher("* equals(java.lang.Object, java.lang.Object)");
+    private static final MethodMatcher HASHCODE = new MethodMatcher("* hashCode(java.lang.Object)");
+    private static final MethodMatcher NULL_SAFE_GET_STRING_ARRAY = new MethodMatcher("* nullSafeGet(java.sql.ResultSet, java.lang.String[], org.hibernate.engine.spi.SharedSessionContractImplementor, java.lang.Object)");
+    private static final MethodMatcher NULL_SAFE_GET_INT = new MethodMatcher("* nullSafeGet(java.sql.ResultSet, int, org.hibernate.engine.spi.SharedSessionContractImplementor, java.lang.Object)");
+    private static final MethodMatcher RESULT_SET_STRING_PARAM = new MethodMatcher("java.sql.ResultSet *(java.lang.String)");
+    private static final MethodMatcher NULL_SAFE_SET = new MethodMatcher("* nullSafeSet(java.sql.PreparedStatement, java.lang.Object, int, org.hibernate.engine.spi.SharedSessionContractImplementor)");
+    private static final MethodMatcher DEEP_COPY = new MethodMatcher("* deepCopy(java.lang.Object)");
+    private static final MethodMatcher DISASSEMBLE = new MethodMatcher("* disassemble(java.lang.Object)");
+    private static final MethodMatcher ASSEMBLE = new MethodMatcher("* assemble(java.io.Serializable, java.lang.Object)");
+    private static final MethodMatcher REPLACE = new MethodMatcher("* replace(java.lang.Object, java.lang.Object, java.lang.Object)");
+
+    @Nullable
+    private static J.FieldAccess parameterizedType = null;
+
+    @Override
+    public String getDisplayName() {
+        return "Migrate UserType to hibernate 6";
+    }
+
+    @Override
+    public String getDescription() {
+        return "With hibernate 6 the UserType interface received a type parameter making it more strictly typed" +
+               "This recipe applies the changes required to adhere to this change.";
```

Comment:
Be careful when you join strings; this would look odd in the docs without `. ` in between.
```suggestion
        return "With hibernate 6 the UserType interface received a type parameter making it more strictly typed. " +
               "This recipe applies the changes required to adhere to this change.";
```

Reply:
I can't recall doing this intentionally but I'll keep it in mind!